shell/platform/embedder/embedder.cc

```diff
@@ -814,7 +814,11 @@ FlutterEngineResult FlutterEngineSendPlatformMessage(
     return LOG_EMBEDDER_ERROR(kInvalidArguments);
   }
 
-  if (SAFE_ACCESS(flutter_message, channel, nullptr) == nullptr ||
+  if (SAFE_ACCESS(flutter_message, channel, nullptr) == nullptr) {
+    return LOG_EMBEDDER_ERROR(kInvalidArguments);
+  }
+
+  if (SAFE_ACCESS(flutter_message, message_size, -1) != 0 &&
```

Comment:
Why are you providing `-1` as the default value rather than `0`? If that path were triggered, then this condition would return false and the check wouldn't work as expected (and then later code would read garbage when accessing the message and message size).